In [1]:
1

[1] 1

In [ ]:
x   ---  gfdgdgfdgdsgdgdgdgdfdwedwed

Aqui pueden escribir lo que sea.
Celdas Markdown nos sirven para ir comentando o describiendo lo que vamos haciendo o programando

In [2]:
# Esto es un comentario
5+7

[1] 12

Operaciones aritméticas

In [3]:
# Multiplicaciones *
150005 * 546584

[1] 81990332920

In [4]:
# divisiones
4 / 2

[1] 2

In [5]:
# exponenciar
3 ** 2

[1] 9

In [6]:
3 ^ 2

[1] 9

El operador residuo

$$
\frac{a}{b}=q
$$

un residuo $r$

In [7]:
# operador residuo %%
4 %% 2

[1] 0

In [8]:
5 %% 4

[1] 1

Variables

In [9]:
3 ** 2

[1] 9

In [10]:
# <- : el operador de asignacion
x <- 3 ** 2

In [14]:
y <- 45

In [13]:
3 * x

[1] 27

In [15]:
# 9 / 45
x / y

[1] 0.2

In [16]:
numero pi <- 3.1416

ERROR: Error in parse(text = x, srcfile = src): <text>:1:8: unexpected symbol
1: numero pi
           ^


In [18]:
# "pi" R nos arroja el valor del numero pi
pi

[1] 3.141593

In [19]:
# 
x <- 3 ** 2

In [20]:
x

[1] 9

Calcular el área de un círculo

In [2]:
numero_pi <- pi
# Definimos una variable que simule el radio de un circulo
r <- 4

pi * r ** 2

[1] 50.26548

In [3]:
r

[1] 4

In [4]:
r <- 7

In [5]:
r

[1] 7

In [6]:
pi * r ** 2

[1] 153.938

En R:

<-: asignación

=: asignación (no confundir que = significa igual)

son equivalentes.

La idea de igualdad que conocemos la rescatamos con el símbolo ==, sí es igualdad

In [9]:
# asigno
i = 4
# obtener el valor de esa variable
i

ERROR: Error in eval(expr, envir, enclos): object 'i' not found


números enteros: números enteros (int)
números decimales: (numeric)
booleanos: TRUE (verdadero) y FALSE (falso)


Otro tipo de dato: Vectores

$$
(5,8,2,1)
$$


El tipo de dato de un vector tiene ser el mismo. 

En principio no podemos combinar.

totalmente distinto (8,5,2,1)

In [14]:
# Vector numeros c()--------> ()
vector1=c(5,8,2,1)
vector1

[1] 5 8 2 1

In [16]:
# arroja el numero de elementos de un vector
length(vector1)

[1] 4

In [15]:
# vector decimal o del tipo numeric
vector2 = c(5.3, 8.0, 2.1, pi)
vector2

[1] 5.300000 8.000000 2.100000 3.141593

In [17]:
length(vector2)

[1] 4

Operaciones entre vectores se realiza entrada a entrada:

In [18]:
vector1 +  vector2

[1] 10.300000 16.000000  4.100000  4.141593

In [19]:
vector1 *  vector2

[1] 26.500000 64.000000  4.200000  3.141593

In [28]:
vector1 **  vector2

[1] 4.181955e+03 1.677722e+07 4.410000e+00 3.141593e+00

In [29]:
# Esto el residuo de las divisiones entrada a entrada
vector2 %%  vector1

[1] 0.3000000 0.0000000 0.1000000 0.1415927

In [30]:
7 %% 2 

[1] 1

In [31]:
ganancias <- c(20, 50, 30, 60)
# sum: suma todas entradas de un vector
sum(ganancias)

[1] 160

In [32]:
# Promedio de un vector
mean(ganancias)

[1] 40

In [33]:
sort(ganancias)

[1] 20 30 50 60

In [38]:
sort(ganancias, decreasing = FALSE)

[1] 20 30 50 60

los boleanos deben ir en mayúsculas

Verdadero : TRUE (T)
Falso: FALSE (F)

In [37]:
#     vector,   decreasing =
sort(ganancias, decreasing = TRUE)

[1] 60 50 30 20

In [42]:
x <- 3 ** 2
x

[1] 9

In [43]:
append(ganancias, 150)

[1]  20  50  30  60 150

In [44]:
# Agregar elementos nuevos a un vector: append()
#        vector 
ganancias <- append(ganancias, 150)
ganancias

[1]  20  50  30  60 150

In [45]:
ganancias <- append(ganancias, 200)
ganancias

[1]  20  50  30  60 150 200

[1]   5  50  30  60 150 200

Jerarquía de tipos de datos:

* Numeric
* Integer

2 --> 2.0
integer --> numeric


(2, 3.14) --> (2.0, 3.14)

Generar secuencias

In [48]:
# primer numero de la secuencia: 1
# ultimo numero de la secuencia: 100
# 1,2,3,...,100
vector3 <- c(1:100)
vector3

[1]   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18
 [19]  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36
 [37]  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54
 [55]  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72
 [73]  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90
 [91]  91  92  93  94  95  96  97  98  99 100

In [49]:
vector4 <- 50:500
vector4

[1]  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67
 [19]  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85
 [37]  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103
 [55] 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121
 [73] 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139
 [91] 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157
[109] 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175
[127] 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193
[145] 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211
[163] 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229
[181] 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247
[199] 248 249 250 251 252 253 254 255 256 257 258 259 260 261 262 263 264 265
[217] 266 267 268 269 270 271 272 273 274 275 276 277 278 279 280 281 282 283
[235] 284 285 286 287 288 289 290 291 292 293 294 295 296 297 298 299 300 301
[253] 302 303 304 305 306 307 308 309 310 311 312 313 314 315 316 317 318 319
[271] 320 321 322 323 324 325 326 327 328 329 330 331 332 333 334 335 336 337
[289] 338 339 340 341 342 343 344 345 346 347 348 349 350 351 352 353 354 355
[307] 356 357 358 359 360 361 362 363 364 365 366 367 368 369 370 371 372 373
[325] 374 375 376 377 378 379 380 381 382 383 384 385 386 387 388 389 390 391
[343] 392 393 394 395 396 397 398 399 400 401 402 403 404 405 406 407 408 409
[361] 410 411 412 413 414 415 416 417 418 419 420 421 422 423 424 425 426 427
[379] 428 429 430 431 432 433 434 435 436 437 438 439 440 441 442 443 444 445
[397] 446 447 448 449 450 451 452 453 454 455 456 457 458 459 460 461 462 463
[415] 464 465 466 467 468 469 470 471 472 473 474 475 476 477 478 479 480 481
[433] 482 483 484 485 486 487 488 489 490 491 492 493 494 495 496 497 498 499
[451] 500

In [54]:
# avance de dos en dos
vector5 <- seq(1,100, by=2)
vector5

[1]  1  3  5  7  9 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49
[26] 51 53 55 57 59 61 63 65 67 69 71 73 75 77 79 81 83 85 87 89 91 93 95 97 99

In [52]:
# avance decimal
vector5 <- seq(1,10, by=0.1)
vector5

[1]  1.0  1.1  1.2  1.3  1.4  1.5  1.6  1.7  1.8  1.9  2.0  2.1  2.2  2.3  2.4
[16]  2.5  2.6  2.7  2.8  2.9  3.0  3.1  3.2  3.3  3.4  3.5  3.6  3.7  3.8  3.9
[31]  4.0  4.1  4.2  4.3  4.4  4.5  4.6  4.7  4.8  4.9  5.0  5.1  5.2  5.3  5.4
[46]  5.5  5.6  5.7  5.8  5.9  6.0  6.1  6.2  6.3  6.4  6.5  6.6  6.7  6.8  6.9
[61]  7.0  7.1  7.2  7.3  7.4  7.5  7.6  7.7  7.8  7.9  8.0  8.1  8.2  8.3  8.4
[76]  8.5  8.6  8.7  8.8  8.9  9.0  9.1  9.2  9.3  9.4  9.5  9.6  9.7  9.8  9.9
[91] 10.0

Accediendo a los elementos de un vector

In [64]:
x <- 3 ** 2

In [65]:
x

[1] 9

In [62]:
ganancias

[1]   5  50  30  60 150 200

In [56]:
# nombre del vector, seguido de [], dentro de los
# corchetes colocan la posicion
ganancias[1]

[1] 5

In [57]:
ganancias[5]

[1] 150

In [66]:
length(ganancias)

[1] 6

In [58]:
ganancias[100]

[1] NA

In [59]:
# 1:3 ---> 1,2,3
ganancias[1:3]

[1]  5 50 30

In [60]:
ganancias[c(1,2,3)]

[1]  5 50 30

In [61]:
ganancias[c(3, 1, 5)]

[1]  30   5 150

In [68]:
ganancias

[1]   5  50  30  60 150 200

Agregar elemento nuevo en la posición 3

In [71]:
ganancias[3] <- 60

In [72]:
ganancias

[1]   5  50  60  60 150 200

En R las numeraciones comienzan en 1:

In [78]:
s <- 5
s

[1] 5

In [81]:
# Eliminar variables rm()
rm(s)

Warning message in rm(s):
“object 's' not found”


In [82]:
s

ERROR: Error in eval(expr, envir, enclos): object 's' not found
